File: src/TaylorBot.Net/src/TaylorBot.Net.EntityTracker.Infrastructure/Username/UsernameRepository.cs
using Dapper;
using Discord;
using Microsoft.Extensions.Options;
using System.Threading.Tasks;
using TaylorBot.Net.Core.Infrastructure;
using TaylorBot.Net.Core.Infrastructure.Options;
using TaylorBot.Net.EntityTracker.Domain.Username;

namespace TaylorBot.Net.EntityTracker.Infrastructure.Username
{
    public class UsernameRepository : PostgresRepository, IUsernameRepository
    {
        public UsernameRepository(IOptionsMonitor<DatabaseConnectionOptions> optionsMonitor) : base(optionsMonitor)
        {
        }

        public async ValueTask AddNewUsernameAsync(IUser user)
        {
            using var connection = Connection;

            await connection.ExecuteAsync(
                "INSERT INTO users.usernames (user_id, username) VALUES (@UserId, @Username);",
                new
                {
                    UserId = user.Id.ToString(),
                    Username = user.Username
                }
            );
        }
    }
}

File: src/taylorbot-typescript/database/repositories/UsernameRepository.ts
import { Log } from '../../tools/Logger';
import { Format } from '../../modules/discord/DiscordFormatter';
import * as pgPromise from 'pg-promise';
import { User } from 'discord.js';

export class UsernameRepository {
    readonly #db: pgPromise.IDatabase<unknown>;
    constructor(db: pgPromise.IDatabase<unknown>) {
        this.#db = db;
    }

    async addNewUsernameAsync(user: User): Promise<void> {
        try {
            await this.#db.none(
                `INSERT INTO users.usernames (user_id, username) VALUES ($[user_id], $[username]);`,
                {
                    user_id: user.id,
                    username: user.username
                }
            );
        }
        catch (e) {
            Log.error(`Adding new username for ${Format.user(user)}: ${e}`);
            throw e;
        }
    }
}
